test_echo.py

```diff
@@ -7,15 +7,19 @@
 
 def test_connection():
     """Test that a connect is establised by reading a returned value"""
-    assert "test" == echo_client("test")
+    response = echo_client("GET test/test/test HTTP/1.1")
+    print response
+    assert "test/test/test" in response
 
 def test_input():
-    """Test that a unicode imput returns correctly"""
-    encoded_text = u'kjfk'
-    assert 'kjfk' == echo_client(encoded_text)
+    """Test that a none GET method returns the correct error"""
+    encoded_text = u'Post test/test/test HTTP/1.1'
+    response = echo_client(encoded_text)
+    print response
+    assert '405' in response
 
 def test_longer_string():
-    """Test that a string over 32 bytes gets read correctly"""
-    print sys.getsizeof('aklsjflksjflkasjflsjflkjlkajflksjdflkasjflkjslkdfjalfalsdfjkl')
-    assert 'aklsjflksjflkasjflsjfl' == echo_client('aklsjflksjflkasjflsjfl')
-
+    """Test that a incorrect http returns the correct error"""
+    response = echo_client('GET test/test/test HTTP/1.2')
+    print response
+    assert '505' in response
```

Comment:
These tests are good, so far as they go.  Worth noting that they are **unit** tests, not functional.  You aren't engaging the end-to-end operation of the server, but just calling a Python function.